app/models/pawn.rb

```diff
@@ -1,5 +1,76 @@
 class Pawn < Piece
 
+  def valid_move?(new_x, new_y)
+    on_board?(new_x, new_y) && passes_pawn_rules?(new_x, new_y)
+  end
+
+  def passes_pawn_rules?(x, y)
+    if two_squares_forward?(x, y)
+      return true
+    elsif one_square_forward?(x, y)
+      return true
+    elsif diagonal_move?(x, y)
+      return true
+    end
+
+    return false
+  end
+
+  def two_squares_forward?(x, y)
+    first_move_position? && not_backwards?(x, y) && (self.y_cord - y).abs == 2 && self.x_cord - x == 0
+  end
+
+  def one_square_forward?(x, y)
+    not_backwards?(x, y) && (self.y_cord - y).abs == 1 && self.x_cord - x == 0
+  end
+
+  def diagonal_move?(x, y)
+    diagonal_allowed?(x, y) && not_backwards?(x, y) && (self.y_cord - y).abs == 1 && (self.x_cord - x).abs == 1
+  end
+
+  #Checks that square is on board
+  def on_board?(x, y)
+    x >= 0 && x <= 7 && y >=0 && y <= 7
+  end
+
+  #Checks that pawn is on the originating square aka first_move_position depending on its color
+  def first_move_position?
```

Comment:
What if we decide to refactor it later and black and white changes its positions for example? This piece of code will need to change.. Is there a possibility to add a flag on a piece if it was moved? Then the whole thing become: 

def moved?
    moved
end